### DISCUSSION TITLE
How to access nuxt.config.ts keys in server api and server middleware?

### DISCUSSION THREAD

**Discussion by BananaAcid** [nuxt/nuxt#20554] [Questions] [answered]
I would like to add keys with my own configuration stuff.

Is there a way to access these keys?

I am not sure, if it is documented some where and I just didn't see it - feel free to link it.

**Comment by warflash** [comment]
I'm not 100% sure what you're refering to with keys here but https://nuxt.com/docs/guide/going-further/runtime-config#exposing-runtime-config might be what you're looking for

**Comment by BananaAcid** [comment]
Thanks @warflash, but that seems not to access a custom key in `nuxt.config.ts`.

An example of what I am trying to access:

nuxt.config.ts
```ts
 export default defineNuxtConfig({
    ssr: true,
    modules: [],
    
    myCustomKey: {
        myCustomUrl: 'https://server.tld/api/v1/'
  });
```

... I would like to access this config info somewhere in my `/server/api/checkConnection.get.ts`

How to plugins access their config property within in `nuxt.config`?

**Reply by warflash** [reply, marked as answer]
Modules access config keys through defineNuxtModule from nuxt/kit. That most certainly is not what you're looking for though.
I'm still not sure what speaks against using runtimeConfig?

```js
export default defineNuxtConfig({
  runtimeConfig: {
    myCustomUrl: 'https://server.tld/api/v1/'
  }
})
```

and then in `/server/api/checkConnection.get.ts` just
```js
const runtimeConfig = useRuntimeConfig()
const url = runtimeConfig.myCustomUrl
```

**Reply by BananaAcid** [reply]
I thought about avoiding all that logic behind RuntimeConfig, like envs/public and more, and just globally store a few protected constants for backend usage only (instead of importing a config.json in each file, or loading the `config.json` into  `globalThis.myConfig`).
